plugins/item_previews/web_client/views/ItemPreviewWidget.js

```diff
@@ -1,103 +1,175 @@
+        return /(json)$/i.test(str);
     },
 
-    _isJSONItem: function (item) {
-        return /\.(json)$/i.test(item.get('name'));
+    _isImageItem: function (str) {
+        return /(jpg|jpeg|png|gif)$/i.test(str);
     },
 
-    _isImageItem: function (item) {
-        return /\.(jpg|jpeg|png|gif)$/i.test(item.get('name'));
+    setElement: function ($el) {
+        this.initialized = false;
+        View.prototype.setElement.apply(this, arguments);
     },
 
     render: function () {
-        var supportedItems = this.collection.filter(this._isSupportedItem.bind(this));
-
-        var view = this;
-
-        view.$el.html(ItemPreviewWidgetTemplate({
-            items: supportedItems,
-            wrapPreviews: view.wrapPreviews,
-            hasMore: this.collection.hasNextPage(),
-            isImageItem: this._isImageItem,
-            isJSONItem: this._isJSONItem
-        }));
-
-        // Render any JSON files.
-        supportedItems.filter(this._isJSONItem).forEach(function (item) {
-            var id = item.get('_id');
-
-            // Don't process JSON files that are too big to preview.
-            var size = item.get('size');
-            if (size > this._MAX_JSON_SIZE) {
-                return view.$('.json[data-id="' + id + '"]').text('JSON too big to preview.');
-            }
+        if (!this.initialized) {
+            this.initialized = true;
+            this.renderedIndex = 0;
+            this.$el.html(ItemPreviewWidgetTemplate(this));
+            this.$('.g-widget-item-previews-container').on('scroll', (e) => this.onScroll(e));
+            this.debouncedAddMoreItem();
+        }
+    },
+
+    isNotFull: function () {
+        return this.$('.g-widget-item-prevews-wrapper').height() <= this.$('.g-widget-item-previews-container').height();
+    },
+
+    allItemAdded: function () {
+        return this.renderedIndex === this.supportedItems.length;
+    },
+
+    addMoreItem: function () {
+        this.addingMoreItems = true;
+        var items = this.supportedItems.slice(this.renderedIndex, this.renderedIndex + this._LOAD_BATCH_SIZE);
+        this.renderedIndex = Math.min(this.renderedIndex + this._LOAD_BATCH_SIZE, this.supportedItems.length);
+
+        if (!items.length) {
+            return;
+        }
 
-            // Ajax request the JSON files to display them.
-            restRequest({
-                path: 'item/' + id + '/download',
-                type: 'GET',
-                error: null // don't do default error behavior (validation may fail)
-            }).done(function (resp) {
-                view.$('.json[data-id="' + id + '"]').text(JSON.stringify(resp, null, '\t'));
-            }).fail(function (err) {
-                console.error('Could not preview item', err);
+        var $container = this.$('.g-widget-item-prevews-wrapper');
+        _whenAll(items.map((item) => {
+            return restRequest({
+                path: `item/${item._id}/files`
```

Comment:
This could potentially queue up 50 files to be downloaded, which would block anything else from loading until it is done.  Maybe use a smaller limit here?